testsuite/tests/lib-systhreads/testpreempt.ml

```diff
@@ -0,0 +1,21 @@
+let rec generate_list n =
+  let rec aux acc = function
+    | 0 -> acc
+    | n -> aux (float n :: acc) (n-1)
+  in
+  aux [] n
+
+let long_computation () =
+  let long_list = generate_list 2000000 in
+  Printf.printf "Long computation result: %d\n%!"
+                (List.length (List.rev_map sin long_list))
+
```

Comment:
Pardon my French but this test is awful.  On a very fast machine (in the future possibly) the `long_computation` may well terminate before 0.2 s causing unexpected output.  On a slow machine or in bytecode the `long_computation` may take a long time and slow down the test suite uselessly.  Please use `Sys.time` to iterate a medium-size computation until 2 seconds are elapsed.

Reply:
I changed the test. The reason I did not do that in the first place is that I wondered that the system calls in `Sys.time` would artificially trigger preemption. But since `Sys.time` does not release the master lock , there is no risk.